### PR TITLE
feat(demo): prevent demo users from seeing login page

### PR DESCRIPTION
**Pull request by scefali**
Demo users who end up in a logged out state after their org/user gets deleted get redirected to the login page. However, this is confusing to users because they won't have a way of logging back in. In the long term, we are planning on building a dedicated screen to handle this case where the user can start a new demo with a new organization. However, right now we are just going to redirect users to the marketing home page since we don't want to build that view right now.

### CODE REVIEW

File: src/sentry/demo/middleware.py [hunk 1]
```diff
@@ -1,33 +1,53 @@
...
         # disable org creation
-        if request.path == org_creation_route and request.method == "POST":
+        if path == org_creation_route and method == "POST":
             return JsonResponse(
                 {"detail": "Organization creation disabled in demo mode"}, status=400
             )
 
-        # only handling org views
+        # backdoor to allow logins
+        allow_login = request.GET.get("allow_login") != "1"
```
Comment (wedamija) [inline src/sentry/demo/middleware.py:34]: Is this backwards? Seems weird that `allow_login != "1"` maps to the inverse.

Comment (scefali) [reply src/sentry/demo/middleware.py:34]: lol yea you're right

Comment (scefali) [reply src/sentry/demo/middleware.py]: @wedamija the logic was actually right but the variable name didn't make sense. Updated.

[the remark this answers is not in the document]
Comment (scefali) [reply src/sentry/demo/middleware.py]: @markstory added

**Review by wedamija — APPROVED** [review]

**Pull request merged**